docs/design/kademlia-removal.md

```diff
@@ -2,88 +2,113 @@
-- Storage nodes should notify satellites when they start up, wait a random amount of time (to add jitter 
-http://highscalability.com/blog/2012/4/17/youtube-strategy-adding-jitter-isnt-a-bug.html), then start reporting in roughly on the hour
-
-### Network refreshes at a regular interval
-- Nodes will keep themselves up to date in the network by pinging all the satellites in their
-   trusted list every hour.
-- Satellites will ping the nodes back to confirm their addresses
-    - If is it successful, the satellite will insert or update the node in the overlay cache and
-       notify the node of success. Make sure to close the connection. Don’t use the transport observer to update the cache.
-       Update the IP and uptime directly.
-    - If the satellite does not confirm the node address, it does not proceed with updating the overlay cache. The node 
-    receives a log message and closes the connection when it times out.
-
-### Disintegrate Kademlia from the network, storj sim and testplanet setups
-- Remove kademlia from the discovery package
-- Remove the bootstrap node - work with Ops
-  - Remove the vouchers service and related tables
-  - Work with QA to make sure storage nodes don’t crash on errors related to the elimination of Kademlia
-- if they don’t update immediately ->  keep just the overlay.Ping rpc method, it will be much easier for a new satellite 
-to work with old and new storage nodes.
-
-### Update whitepaper to address kademlia removal and the addition of satellite opt-in
-  - Delete the audit gating design doc
-  - Update the wiki
+To replace Kademlia, we need to complete several things:
+
+- replace storage node initial communication with the satellites,
+- replace network refreshing,
+- remove kademlia from services,
+- update documentation.
+
+
+### Storage Node initial communication
+
+Storage node should connect satellites in their trusted list and notify that they want to work with them.
+
+We need to ensure that we do not overload the satellite during upgrades.
+Hence, we need to add jitter for refreshes and initial communication.
+
+_For more information on jitter see http://highscalability.com/blog/2012/4/17/youtube-strategy-adding-jitter-isnt-a-bug.html ._
+
+### Network refreshing
+
+Storage Nodes keep themselves up to date in the network by pinging all the satellites in their trusted list. Refreshes would happen every 1 hour.
+Satellites, in response, will ping the nodes to confirm their address and ensure that the network is configured correctly.
+
+When a Satellite has successfully pinged the storage node, it will update IP and uptime in overlay.
+On failure, the satellite does not update overlay and notifies the storage node.
+
+Storage Node keeps track of this information, such that Storage Node operator can notice the problem.
+
+We consider a successful ping when a node with node-ID `N` has contacted satellite `S` and claimed its address is `A`,
```

Comment:
I like this part!

Reply:
Although, does this apply to regular grpc communication both directions (we're punting on the UDP option)

Reply:
This line (52) leaves the method of contact unspecified, because that is where the UDP option would take effect, if I understand it right. When `N` contacts `S` and claims address `A`, it might do that via UDP to a "tip line" or via normal grpc or something else.

The following lines describing pingback could certainly be accomplished by way of GRPC (that would set up the connection and the SSL session). I just didn't want to imply it's the only option.